dxp/liferay_component.md

```diff
@@ -0,0 +1,53 @@
+# Liferay.component()
+
+Liferay.component() registers a component and retrieves its instance from the global registry.
+
+## How to use it
+
+### Register
+
+`Liferay.component(id, value, [componentConfig])`
+
+To register a component is needed to pass an `id` to identify the instance, the component, and as an optional parameter, a custom component configuration to provide additional hints for the system handling of the component lifecycle.
+
+This will return the passed value.
+
+When the component is registered a global Liferay event `id + :registered` is fired, which can be listen with `Liferay.on()` o `Liferay.once()`.
+
+⚠️ WARNING: If there's already a component registered with the passed `id` it will be replaced with the new value, which could lead to unexpected behaviour in the "Liferay.component" and "Liferay.componentReady" APIs, as well as in the `*:registered` events.
+
+### Retrieve
+
+`Liferay.component(id)`
+
+Once the component is registered it can be retrieved by passing to the method the id it was registered with.
+
+One thing to take into account is that if the registered value was a function the first time the component is retrieved the function will be executed and its result stored as the new registered value, which will be returned.
+
+### Notes
+
+Using `<react:component>` tag a component will be registered using `Liferay.Component` but it will not be the react component but a synthetic one, that's why even when a function component is being registered (which doesn't create an instance) a component instance is returned.
+
+# Liferay.componentReady()
+
+`Liferay.componentReady()` retrieves a list of component instances after they've been registered and returns a `Promise` to be resolved with all the requested component instances after they've been successfully registered.
+
+## How to use it
+
+`Liferay.componentReady([...id]).then(function)`
+
+A common use case is the need of run some actions after a component which is not in scope is created.
+
+# Liferay.destroyComponent()
+
+This is the safest way to destroy a registered component. Destroys the component registered by the provided component ID. This invokes the component's own destroy lifecycle methods (destroy or dispose) and deletes the internal references to the component in the component registry.
```

Comment:
"Destroys" → "It destroys"

"registered by" → "registered with"

Backticks around `destroy` and `dispose`